UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -258,6 +258,8 @@ public void PutObservations(string brainKey, ICollection<Agent> agents)
             {
                 foreach (var agent in m_CurrentAgents[brainKey])
                 {
+                    // Update the sensor data on the AgentInfo
+                    agent.GenerateSensorData();
```

Comment:
I think CollectObservation will eventually go away. This is currently here instead of the general agent step because we don't want it called during inferrence (we want to write directly to the tensors then).